engine/src/main/java/org/hibernate/search/bridge/builtin/ArrayBridge.java

```diff
@@ -43,8 +45,9 @@ public FieldBridge getElementBridge() {
 	}
 
 	private void indexNotNullArray(String name, Object value, Document document, LuceneOptions luceneOptions) {
-		Object[] collection = (Object[]) value;
-		for ( Object entry : collection ) {
+		// Use Array.* to also support arrays of primitive values
+		for ( int i = 0, length = Array.getLength( value ) ; i < length ; ++i ) {
```

Comment:
I would vote for not using `Array` and introduce what's needed in `CollectionHelper` as we did for HV.